packages/core/src/repository/query/network-query.ts

```diff
@@ -0,0 +1,20 @@
+import { ParameterType } from '../../data';
+
+export abstract class NetworkQuery {
+    constructor(public readonly endpoint: string) {}
+
+    get body(): string | FormData {
+        return '';
+    }
+    get urlParameters(): Record<string, ParameterType> {
+        return {};
+    }
+    get queryParameters(): Record<string, ParameterType> {
+        return {};
+    }
+}
+
+export class GetNetworkQuery extends NetworkQuery {}
+export class PostNetworkQuery extends NetworkQuery {}
+export class PutNetworkQuery extends NetworkQuery {}
+export class DeleteNetworkQuery extends NetworkQuery {}
```

Comment:
The http request methods are not added since the query class have it implicit. Actually, the actions would have it implicitely included but we need Post and Put network queries to distinguis these methods in the Put method.